### PR TITLE
Create deployment.spatialAnalysis.template.json

### PR DESCRIPTION
**Pull request by naiteeks**

### CODE REVIEW

**Comment by naiteeks** [comment]
@anilmur , @avkewalr-msft  - please review.

File: src/edge/deployment.spatialAnalysis.template.json [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+						"version": "1.0",
+						"type": "docker",
+						"status": "running",
+						"restartPolicy": "always",
+						"settings": {
+							"image": "mcr.microsoft.com/lva-utilities/rtspsim-live555:1.2",
+							"createOptions": {
+								"HostConfig": {
+									"Mounts": [
+										{
```
Comment (anilmur) [inline src/edge/deployment.spatialAnalysis.template.json:95]: Why is this using Mounts for rtspsim but Binds for lvaEdge? Is that okay?

File: src/edge/deployment.spatialAnalysis.template.json [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+								"HostConfig": {
+									"PortBindings": {
+										"50051/tcp": [
+											{
+												"HostPort": "50051"
+											}
+										]
+									},
+									"IpcMode": "host",
+									"Binds": [
```
Comment (anilmur) [inline src/edge/deployment.spatialAnalysis.template.json:129]: Why Binds here and not Mounts

File: src/edge/deployment.spatialAnalysis.template.json (hunk 1 above)
Comment (naiteeks) [reply src/edge/deployment.spatialAnalysis.template.json:95]: Volume mounts are needed for the video sample to be used

File: src/edge/deployment.spatialAnalysis.template.json [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+				"aadServicePrincipalAppId": "$AAD_SERVICE_PRINCIPAL_ID",
+				"aadServicePrincipalSecret": "$AAD_SERVICE_PRINCIPAL_SECRET",
+				"aadEndpoint": "https://login.microsoftonline.com",
+				"aadResourceId": "https://management.core.windows.net/",
+				"armEndpoint": "https://management.azure.com/",
+				"diagnosticsEventsOutputName": "AmsDiagnostics",
+				"operationalEventsOutputName": "AmsOperational",
+				"logLevel": "Info",
+				"logCategories": "Application,Events,MediaPipeline",
+				"allowUnsecuredEndpoints": true,
```
Comment (anilmur) [inline src/edge/deployment.spatialAnalysis.template.json:199]: Why is MediaPipeline included in logCategories?

File: src/edge/deployment.spatialAnalysis.template.json [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+											"max-size": "10m",
+											"max-file": "10"
+										}
+									},
+									"Binds": [
+										"$OUTPUT_VIDEO_FOLDER_ON_DEVICE:/var/media/",
+										"$APPDATA_FOLDER_ON_DEVICE:/var/lib/azuremediaservices"
+									],
+									"IpcMode": "host",
+									"ShmSize": 1536870912
```
[the remark this answers is not in the document]
Comment (naiteeks) [reply src/edge/deployment.spatialAnalysis.template.json:75]: they will get an insufficent memory error. They can overcome the error by decreasing the ShmSize.

File: src/edge/deployment.spatialAnalysis.template.json [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+							},
+							"QT_X11_NO_MITSHM": {
+								"value": "1"
+							},
+							"OMP_WAIT_POLICY": {
+								"value": "PASSIVE"
+							},
+							"EULA": {
+								"value": "accept"
+							},
```
Comment (anilmur) [inline src/edge/deployment.spatialAnalysis.template.json:162]: Did you check with RTCV team and CELA about this? Okay to have EULA to have a default of "accept"?

Comment (naiteeks) [reply src/edge/deployment.spatialAnalysis.template.json:162]: that is a requirement by them

File: src/edge/readme.md [hunk 6]
```diff
@@ -23,13 +12,28 @@ In addition to the modules defined in deployment.template.json, this deployment
...
+In addition to the modules defined in deployment.template.json, this deployment manifest template includes the [Computer Vision for spatial analysis AI module from Azure Cognitive Services](https://azure.microsoft.com/services/cognitive-services/computer-vision/). This module supports operations that enable you to count people in a designated zone within the camera’s field of view, to track when a person crosses a designated line or area, or when people violate a distance rule. This template is used in [this](https://aka.ms/lva-spatial-analysis) tutorial.
+
+### deployment.template.json
+
```
Comment (anilmur) [inline src/edge/readme.md:28]: Did you see how every other paragraph starts with "In addition to the modules..."? This makes sense only when deployment.template.json is the first entry in this readme.

Why did you disrupt the previous order?

File: src/edge/readme.md [hunk 7]
```diff
@@ -23,13 +12,28 @@ In addition to the modules defined in deployment.template.json, this deployment
...
+In addition to the modules defined in deployment.template.json, this deployment manifest template includes the [Computer Vision for spatial analysis AI module from Azure Cognitive Services](https://azure.microsoft.com/services/cognitive-services/computer-vision/). This module supports operations that enable you to count people in a designated zone within the camera’s field of view, to track when a person crosses a designated line or area, or when people violate a distance rule. This template is used in [this](https://aka.ms/lva-spatial-analysis) tutorial.
```
Comment (anilmur) [inline src/edge/readme.md:25]: URL is wrong. You should point to their page for the Spatial Analysis container

**Review by anilmur — CHANGES_REQUESTED** [review]
Some comments, and a few corrections needed

File: src/edge/readme.md (hunk 6 above)
Comment (naiteeks) [reply src/edge/readme.md]: put it back on top. I had re-arranged them as be in alphabetical order as Github shows them in that order.

File: src/edge/deployment.spatialAnalysis.template.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (avkewalr-msft) [reply src/edge/deployment.spatialAnalysis.template.json:75]: naiteek - Is there any documentation on min size needed?
anil - what device do you have in mind?

File: src/edge/deployment.spatialAnalysis.template.json [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+							"BILLING_ENDPOINT": {
+								"value": "<Use one key from Archon azure resource (keys page)>"
```
Comment (avkewalr-msft) [inline src/edge/deployment.spatialAnalysis.template.json:164]: where does this value come from? I did not see the discussion in the tutorial

Comment (naiteeks) [reply src/edge/deployment.spatialAnalysis.template.json:164]: https://docs.microsoft.com/en-us/azure/media-services/live-video-analytics-edge/spatial-analysis-tutorial#endpoint-uri-endpoint_uri

File: src/edge/deployment.spatialAnalysis.template.json (hunk 3 above)
Comment (naiteeks) [reply src/edge/deployment.spatialAnalysis.template.json:199]: that was needed to get additional logs. But i have removed that now.

File: src/edge/deployment.spatialAnalysis.template.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (naiteeks) [reply src/edge/deployment.spatialAnalysis.template.json:75]: we dont have a min size needed documented anywhere yet.

**Pull request merged**